File: ruby/lesson 8/dfs_graph.rb
def traverse(graph)
    visited = Array.new(graph.size)
    (0..graph.size - 1).each { |v| dfs(graph, v, visited) }
end

def dfs(graph, v, visited)
    return if visited[v]

    visited[v] = true
    puts(v)

    graph.get(v).each { |w| dfs(graph, w, visited) }
end

File: javascript/lesson 8/dfsGraph.js
function traverse(graph) {
    const visited = Array(graph.size());
    for (let v = 0; v < graph.size(); v++) {
        dfs(graph, v, visited);
    }
}

function dfs(graph, v, visited) {
    if (visited[v]) return; 

    visited[v] = true;
    console.log(v);

    for (let w of graph.get(v)) {
       dfs(graph, w, visited);
    }
}
